### PR TITLE
[12.0][IMP] account: consider account_fiscal_year from OCA/account-financial-tools

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
In case of having that module, it's good to consider it for filling the new model.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by pedrobaeza — APPROVED** [review]
Not using it, but seems correct

**Comment by pedrobaeza** [comment]
I have retargeted the PR but it seems there are conflicts.

**Comment by MiquelRForgeFlow** [comment]
Solved (it was due to the branch retargeting)

File: addons/account/migrations/12.0.1.1/post-migration.py [hunk 1]
```diff
@@ -221,7 +221,28 @@ def set_default_taxes(env):
 
 
 def populate_fiscal_years(env):
-    if openupgrade.table_exists(env.cr, 'account_fiscalyear'):
```
Comment (legalsylvain) [inline addons/account/migrations/12.0.1.1/post-migration.py]: Better to keep this test, and if is false test if date_rangetype.

Otherwise. Lgtm. Thanks.

**Review by legalsylvain — CHANGES_REQUESTED** [review]

File: addons/account/migrations/12.0.1.1/post-migration.py (hunk 1 above)
Comment (MiquelRForgeFlow) [reply addons/account/migrations/12.0.1.1/post-migration.py]: Done

File: addons/account/migrations/12.0.1.1/post-migration.py [hunk 2]
```diff
@@ -235,6 +235,37 @@ def populate_fiscal_years(env):
...
+            )
+            SELECT
+                dr.name, COALESCE(dr.company_id, drt.company_id),
+                dr.date_start, dr.date_end,
+                dr.create_date, dr.create_uid, dr.write_date, dr.write_uid
+            FROM date_range dr
+            JOIN date_range_type drt ON dr.type_id = drt.id
+            WHERE drt.fiscal_year AND COALESCE(dr.company_id, drt.company_id) IS NOT NULL
+            """)
+        for company in env["res.company"].search([]):
```
Comment (pedrobaeza) [inline addons/account/migrations/12.0.1.1/post-migration.py]: Don't do loop but do JOIN with res_company in the SQL query, and please reuse the common query part putting it in a string variable for more readable code.

**Review by legalsylvain — APPROVED** [review]

File: addons/account/migrations/12.0.1.1/post-migration.py (hunk 2 above)
Comment (MiquelRForgeFlow) [reply addons/account/migrations/12.0.1.1/post-migration.py]: How to do the join of res_company?

Comment (pedrobaeza) [reply addons/account/migrations/12.0.1.1/post-migration.py]: With a CROSS JOIN

Comment (MiquelRForgeFlow) [reply addons/account/migrations/12.0.1.1/post-migration.py]: Done :)

**Pull request merged**